### PR TITLE
[Feature][connector-v2]Support opengauss jdbc connnector using opengauss driver.

### PR DESCRIPTION
**Pull request by luckyLJY**
Use opengauss driver to implement opengauss jdbc sink and source

### CODE REVIEW

**Comment by dailai** [comment]
Good job. I often encounter authentication issues when using jdbc-pg to connect to opengauss. When the password_encryption_type=1 of the Opengauss database is used to encrypt the password in the sha256 mode, which is compatible with the MD5 authentication mode of the postgres client， but when password_encryption_type=2 means that the password is encrypted using sha256, which is not compatible with the postgres driver . The postgres driver gives an error,

File: seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/catalog/opengauss/OpenGaussCatalog.java [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+import lombok.extern.slf4j.Slf4j;
+
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+
+@Slf4j
+public class OpenGaussCatalog extends AbstractJdbcCatalog {
+
+    private static final String SELECT_COLUMNS_SQL_TEMPLATE =
```
Comment (hawk9821) [inline seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/catalog/opengauss/OpenGaussCatalog.java]: openguass support `select * FROM information_schema.columns`  ,  advised to use the information_schema.columns view

File: docs/en/connector-v2/sink/Jdbc.md [hunk 2]
```diff
@@ -248,6 +248,7 @@ there are some reference value for params above.
...
 | xugu              | com.xugu.cloudjdbc.Driver                    | jdbc:xugu://localhost:5138                                         | /                                                  | https://repo1.maven.org/maven2/com/xugudb/xugu-jdbc/12.2.0/xugu-jdbc-12.2.0.jar                                               |
 | InterSystems IRIS | com.intersystems.jdbc.IRISDriver             | jdbc:IRIS://localhost:1972/%SYS                                    | /                                                  | https://raw.githubusercontent.com/intersystems-community/iris-driver-distribution/main/JDBC/JDK18/intersystems-jdbc-3.8.4.jar |
+| opengauss         | org.opengauss.Driver                         | jdbc:opengauss://localhost:5432/postgres                           | /                                                  | https://repo1.maven.org/maven2/org/opengauss/opengauss-jdbc/5.1.0-og/opengauss-jdbc-5.1.0-og.jar                              |
```
Comment (hawk9821) [inline docs/en/connector-v2/sink/Jdbc.md:251]: Please refer to the official documentation,  the database name must be  postgres ?  advised to change the database name  [https://docs-opengauss.osinfra.cn/en/docs/5.0.0/docs/DeveloperGuide/connecting-to-a-database-JDBC.html](https://docs-opengauss.osinfra.cn/en/docs/5.0.0/docs/DeveloperGuide/connecting-to-a-database-JDBC.html)

File: docs/en/connector-v2/source/Jdbc.md [hunk 3]
```diff
@@ -137,6 +137,7 @@ there are some reference value for params above.
 | Hive              | org.apache.hive.jdbc.HiveDriver                     | jdbc:hive2://localhost:10000                                           | https://repo1.maven.org/maven2/org/apache/hive/hive-jdbc/3.1.3/hive-jdbc-3.1.3-standalone.jar                                 |
 | xugu              | com.xugu.cloudjdbc.Driver                           | jdbc:xugu://localhost:5138                                             | https://repo1.maven.org/maven2/com/xugudb/xugu-jdbc/12.2.0/xugu-jdbc-12.2.0.jar                                               |
 | InterSystems IRIS | com.intersystems.jdbc.IRISDriver                    | jdbc:IRIS://localhost:1972/%SYS                                        | https://raw.githubusercontent.com/intersystems-community/iris-driver-distribution/main/JDBC/JDK18/intersystems-jdbc-3.8.4.jar |
+| opengauss         | org.opengauss.Driver                                | jdbc:opengauss://localhost:5432/postgres                               | https://repo1.maven.org/maven2/org/opengauss/opengauss-jdbc/5.1.0-og/opengauss-jdbc-5.1.0-og.jar                              |
```
Comment (hawk9821) [inline docs/en/connector-v2/source/Jdbc.md:140]: ditto

File: docs/zh/connector-v2/sink/Jdbc.md [hunk 4]
```diff
@@ -216,26 +216,27 @@ Sink插件常用参数，请参考 [Sink常用选项](../sink-common-options.md)
...
+| Vertica    | com.vertica.jdbc.Driver                      | jdbc:vertica://localhost:5433                                      | /                                                  | https://repo1.maven.org/maven2/com/vertica/jdbc/vertica-jdbc/12.0.3-0/vertica-jdbc-12.0.3-0.jar      |
+| Kingbase   | com.kingbase8.Driver                         | jdbc:kingbase8://localhost:54321/db_test                           | /                                                  | https://repo1.maven.org/maven2/cn/com/kingbase/kingbase8/8.6.0/kingbase8-8.6.0.jar                   |
+| OceanBase  | com.oceanbase.jdbc.Driver                    | jdbc:oceanbase://localhost:2881                                    | /                                                  | https://repo1.maven.org/maven2/com/oceanbase/oceanbase-client/2.4.11/oceanbase-client-2.4.11.jar     |
+| opengauss  | org.opengauss.Driver                         | jdbc:opengauss://localhost:5432/postgres                           | /                                                  | https://repo1.maven.org/maven2/org/opengauss/opengauss-jdbc/5.1.0-og/opengauss-jdbc-5.1.0-og.jar     |
```
Comment (hawk9821) [inline docs/zh/connector-v2/sink/Jdbc.md:239]: ditto

**Comment by luckyLJY** [comment]
Conflict resolved Request merge

File: seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/dialect/opengauss/OpenGaussDialectFactory.java [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+
+    @Override
+    public JdbcDialect create() {
+        throw new UnsupportedOperationException(
+                "Can't create JdbcDialect without compatible mode for OpenGauss");
+    }
+
+    @Override
+    public JdbcDialect create(String compatibleMode, String fieldIde) {
+        if ("postgresLow".equalsIgnoreCase(compatibleMode)) {
```
Comment (hawk9821) [inline seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/dialect/opengauss/OpenGaussDialectFactory.java]: compatibility dialects have been verified,  please add test cases

File: seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/dialect/opengauss/OpenGaussJdbcRowConverter.java [hunk 6]
```diff
@@ -0,0 +1,145 @@
...
+import java.sql.Date;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Time;
+import java.sql.Timestamp;
+import java.util.Locale;
+import java.util.Optional;
+
+public class OpenGaussJdbcRowConverter extends AbstractJdbcRowConverter {
+    private static final String GS_GEOMETRY = "GEOMETRY";
```
Comment (hawk9821) [inline seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/dialect/opengauss/OpenGaussJdbcRowConverter.java]: Please refer to the official documentation  [data-types](https://docs-opengauss.osinfra.cn/en/docs/5.0.0/docs/SQLReference/data-types.html) ， make sure all types in the official documentation are supported

File: seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/dialect/opengauss/OpenGaussTypeConverter.java [hunk 7]
```diff
@@ -0,0 +1,483 @@
...
+import org.apache.seatunnel.common.exception.CommonError;
+import org.apache.seatunnel.connectors.seatunnel.common.source.TypeDefineUtils;
+import org.apache.seatunnel.connectors.seatunnel.jdbc.internal.dialect.DatabaseIdentifier;
+
+import com.google.auto.service.AutoService;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+@AutoService(TypeConverter.class)
+public class OpenGaussTypeConverter implements TypeConverter<BasicTypeDefine> {
```
Comment (hawk9821) [inline seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/dialect/opengauss/OpenGaussTypeConverter.java]: ditto

File: docs/en/connector-v2/sink/Jdbc.md (hunk 2 above)
Comment (luckyLJY) [reply docs/en/connector-v2/sink/Jdbc.md:251]: This is the database name in the image and has no effect on functionality.

File: docs/en/connector-v2/source/Jdbc.md (hunk 3 above)
Comment (luckyLJY) [reply docs/en/connector-v2/source/Jdbc.md:140]: This is the database name in the image and has no effect on functionality.

File: docs/zh/connector-v2/sink/Jdbc.md (hunk 4 above)
Comment (luckyLJY) [reply docs/zh/connector-v2/sink/Jdbc.md:239]: This is the database name in the image and has no effect on functionality.

File: seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/dialect/opengauss/OpenGaussDialectFactory.java (hunk 5 above)
Comment (luckyLJY) [reply seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/dialect/opengauss/OpenGaussDialectFactory.java]: The configuration of this property is already provided in the case conf configuration file.

File: seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/catalog/opengauss/OpenGaussCatalog.java (hunk 1 above)
Comment (luckyLJY) [reply seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/catalog/opengauss/OpenGaussCatalog.java]: select * FROM information_schema.columns， Field comments and default value information cannot be obtained using this statement

File: seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/dialect/opengauss/OpenGaussJdbcRowConverter.java (hunk 6 above)
Comment (luckyLJY) [reply seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/dialect/opengauss/OpenGaussJdbcRowConverter.java]: OK

File: seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/dialect/opengauss/OpenGaussTypeConverter.java (hunk 7 above)
Comment (luckyLJY) [reply seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/dialect/opengauss/OpenGaussTypeConverter.java]: OK

**Comment by dailai** [comment]
This implementation code is almost identical to jdbc-postgresql, so I suggest you just rely on jdbc-postgresql's implementation and override only the implementation of specific methods, such as those involving jdbc-url and the drivers in the pom.

**Comment by luckyLJY** [comment]
> This implementation code is almost identical to jdbc-postgresql, so I suggest you just rely on jdbc-postgresql's implementation and override only the implementation of specific methods, such as those involving jdbc-url and the drivers in the pom.

Make changes to the code as suggested.

File: seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/dialect/opengauss/OpenGaussDialectFactory.java [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+import org.apache.seatunnel.connectors.seatunnel.jdbc.internal.dialect.JdbcDialectFactory;
+import org.apache.seatunnel.connectors.seatunnel.jdbc.internal.dialect.psql.PostgresDialect;
+import org.apache.seatunnel.connectors.seatunnel.jdbc.internal.dialect.psqllow.PostgresLowDialect;
+
+import com.google.auto.service.AutoService;
+
+import javax.annotation.Nonnull;
+
+@AutoService(JdbcDialectFactory.class)
+public class OpenGaussDialectFactory implements JdbcDialectFactory {
```
Comment (dailai) [inline seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/dialect/opengauss/OpenGaussDialectFactory.java]: You can simply inherit from PostgresDialectFactory and override the acceptsURL method

Comment (luckyLJY) [reply seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/dialect/opengauss/OpenGaussDialectFactory.java]: ok

**Review by dailai — APPROVED** [review]
LGTM if ci passed

**Review by Hisoka-X — APPROVED** [review]
Thanks @luckyLJY !

**Comment by Hisoka-X** [comment]
@hawk9821 would you mind take a look again?

**Review by hawk9821 — APPROVED** [review]
LGTM . thanks @luckyLJY

**Pull request merged**